### PR TITLE
Support nested types in ORDER BY clause

### PR DESCRIPTION
**Pull request by lnkuiper**
This PR implements sorting of nested types, e.g.:

```
CREATE TABLE test0 (i INT[]);
INSERT INTO test0 VALUES ([2]), ([1]), ([1, 2]), ([]), ([2, 2]), ([NULL]), (NULL), ([2, 3]);
SELECT * FROM test0 ORDER BY i;
```

Now yields:
```
NULL
[]
[1]
[1, 2]
[2]
[2, 2]
[2, 3]
[NULL]
```

The comparison is recursive i.e. you can order by a list of struct of list of struct etc.

### CODE REVIEW

File: src/common/types.cpp [hunk 1]
```diff
@@ -267,6 +266,10 @@ bool TypeIsInteger(PhysicalType type) {
 	return (type >= PhysicalType::UINT8 && type <= PhysicalType::INT64) || type == PhysicalType::INT128;
 }
 
+bool TypeIsNested(PhysicalType type) {
```
Comment (Mytherin) [inline src/common/types.cpp:269]: This function appears to be unused, is that intended?

File: src/common/types/row_data_collection.cpp [hunk 2]
```diff
@@ -46,12 +55,12 @@ void RowDataCollection::TemplatedSerializeVectorSortable(VectorData &vdata, cons
 	} else {
 		for (idx_t i = 0; i < add_count; i++) {
 			auto idx = sel.get_index(i);
-			auto source_idx = vdata.sel->get_index(idx);
```
Comment (Mytherin) [inline src/common/types/row_data_collection.cpp:49]: The branch with `has_null = false` appears to be untested - could you add a test with a list that involves no null values?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great, and nice tests! Some comments:

File: src/common/types.cpp (hunk 1 above)
Comment (lnkuiper) [reply src/common/types.cpp:269]: Ah, not that is not intended. I used that shortly during development, but in the end it was more useful to check the physical type directly. I will remove it for now.

File: src/common/types/row_data_collection.cpp (hunk 2 above)
Comment (lnkuiper) [reply src/common/types/row_data_collection.cpp:49]: This was due to not using ValidityStatistics while serializing. This branch should now be used again. Waiting for CI

**Comment by Mytherin** [comment]
Thanks, looks good!

**Pull request merged**